lib/createGltf.js

```diff
@@ -438,38 +423,39 @@ function createSpecularGlossinessTexture(gltf, specularImage, glossinessImage, o
     var height = dimensions[1];
     var pixelsLength = width * height;
     var pixels = Buffer.alloc(pixelsLength * 4, 0xFF); // Initialize with 4 channels, unused channels will be white
+    var scratchChannel = Buffer.alloc(pixelsLength);
 
     if (packSpecular) {
         // Write into the R, G, B channels
-        var redChannel = getImageChannel(specularImage, 0, width, height);
-        var greenChannel = getImageChannel(specularImage, 1, width, height);
-        var blueChannel = getImageChannel(specularImage, 2, width, height);
-        writeChannel(pixels, redChannel, 0, width, height);
-        writeChannel(pixels, greenChannel, 1, width, height);
-        writeChannel(pixels, blueChannel, 2, width, height);
+        var redChannel = getImageChannel(specularImage, 0, width, height, scratchChannel);
+        var greenChannel = getImageChannel(specularImage, 1, width, height, scratchChannel);
+        var blueChannel = getImageChannel(specularImage, 2, width, height, scratchChannel);
+        writeChannel(pixels, redChannel, 0);
+        writeChannel(pixels, greenChannel, 1);
+        writeChannel(pixels, blueChannel, 2);
```

Comment:
Should these be interleaved? Otherwise it seems like this will `writeChannel` the blue channel repeatedly.